internal/config/config.go

```diff
@@ -96,25 +99,52 @@ type RunningPlugin struct {
 	Config *PluginConfig
 }
 
-// PluginConfig containing a name, interval, and drop/pass prefix lists
-// Also lists the tags to filter
-type PluginConfig struct {
-	Name string
-
+// Filter containing drop/pass and tagdrop/tagpass rules
+type Filter struct {
 	Drop []string
 	Pass []string
 
 	TagDrop []TagFilter
 	TagPass []TagFilter
 
+	IsActive bool
+}
+
+// PluginConfig containing a name, interval, and filter
+type PluginConfig struct {
+	Name     string
+	Filter   Filter
 	Interval time.Duration
 }
 
+// OutputConfig containing name and filter
+type OutputConfig struct {
+	Name   string
+	Filter Filter
+}
+
+// Filter returns filtered slice of client.Points based on whether filters
+// are active for this RunningOutput.
+func (ro *RunningOutput) FilterPoints(points []*client.Point) []*client.Point {
+	if !ro.Config.Filter.IsActive {
+		return points
+	}
+
+	var filteredPoints []*client.Point
+	for i := range points {
+		if !ro.Config.Filter.ShouldPass(points[i].Name()) || !ro.Config.Filter.ShouldTagsPass(points[i].Tags()) {
+			continue
+		}
+		filteredPoints = append(filteredPoints, points[i])
+	}
+	return filteredPoints
+}
+
 // ShouldPass returns true if the metric should pass, false if should drop
-// based on the drop/pass plugin parameters
-func (cp *PluginConfig) ShouldPass(measurement string) bool {
-	if cp.Pass != nil {
-		for _, pat := range cp.Pass {
+// based on the drop/pass filter parameters
+func (f Filter) ShouldPass(measurement string) bool {
```

Comment:
Any reason you made these non-pointer receivers? (`Filter` instead of `*Filter`)